html/interaction/focus/the-autofocus-attribute/supported-elements.html

```diff
@@ -34,4 +34,48 @@
   await waitUntilStableAutofocusState(w);
   assert_equals(w.document.activeElement.tagName, 'BODY');
 }, 'Non-HTMLElement should not support autofocus');
+
+promise_test(async t => {
+  let w = window.open('/common/blank.html');
+  await waitForLoad(w);
+  t.add_cleanup(() => { w.close(); });
+  const host = w.document.createElement('div');
+  host.autofocus = true;
+  const shadow = host.attachShadow({mode:'closed', delegatesFocus:true});
+  shadow.appendChild(w.document.createElement('input'));
+  w.document.body.appendChild(host);
+  await waitUntilStableAutofocusState(w);
+  assert_equals(w.document.activeElement, host);
+  assert_equals(shadow.activeElement.tagName, 'INPUT');
+}, 'Host element with delegatesFocus should support autofocus');
+
+promise_test(async t => {
+  let w = window.open('/common/blank.html');
+  await waitForLoad(w);
+  t.add_cleanup(() => { w.close(); });
+  const host = w.document.createElement('div');
+  host.autofocus = true;
+  host.attachShadow({mode:'closed', delegatesFocus:true});
+  w.document.body.appendChild(host);
+  const next = w.document.createElement('input');
+  next.autofocus = true;
+  w.document.body.appendChild(next);
+  await waitUntilStableAutofocusState(w);
+  assert_equals(w.document.activeElement, next);
+}, 'Host element with delegatesFocus including no focusable descendants should be skipped');
+
+promise_test(async t => {
+  let w = window.open('./resources/imagemap.html');
+  await waitForLoad(w);
+  t.add_cleanup(() => { w.close(); });
+  const area = w.document.createElement('area');
+  area.autofocus = true;
+  area.shape = 'rect';
+  area.coords = '1,1,99,99';
+  area.href = '/common/blank.html';
+  w.document.querySelector('map').appendChild(area);
+  await waitUntilStableAutofocusState(w);
+  // Why assert_in_array? See https://github.com/whatwg/html/issues/5054
+  assert_in_array(w.document.activeElement, [area, w.document.querySelector('img')]);
```

Comment:
Hmm, is there any browser where it won't be `area`? I'd prefer to just assert that it's `area`, if all browsers match. Although we'd need to keep the spec issue reference since the spec doesn't match yet.

Reply:
AFAIK, all major browsers set &lt;area> as a DOM anchor.  ok, I updated this so that it asserts only &lt;area>.